src/main/java/org/apache/sysds/runtime/instructions/cp/CovarianceCPInstruction.java

```diff
@@ -82,4 +85,28 @@ public void processInstruction(ExecutionContext ec)
 		double val = covobj.getRequiredResult(_optr);
 		ec.setScalarOutput(output_name, new DoubleObject(val));
 	}
+
+	@Override
+	protected FEDInstruction tryReplaceWithFederated(ExecutionContext ec) {
+		/* NOTE: the requirement before was:
+		if( (instruction.input1.isMatrix() && ec.getMatrixObject(instruction.input1).isFederatedExcept(FTypes.FType.BROADCAST))
+				|| (instruction.input2.isMatrix() && ec.getMatrixObject(instruction.input2).isFederatedExcept(FTypes.FType.BROADCAST))) {
+			if(instruction.getOpcode().equals(...) )
+				...
+			else if("cov".equals(instruction.getOpcode()) && (ec.getMatrixObject(instruction.input1).isFederated(
+					FTypes.FType.ROW) ||
+					ec.getMatrixObject(instruction.input2).isFederated(FTypes.FType.ROW)))
+				...
+			else
+				...
+		
+		This allowed `input1(Broadcast & Row) & input2(!Broadcast & ...)`, I am not sure if that is intentional.
+		 */
+		if((input1.isMatrix() && ec.getMatrixObject(input1).isFederated(FTypes.FType.ROW) &&
+			ec.getMatrixObject(input1).isFederatedExcept(FTypes.FType.BROADCAST)) ||
+			(input2.isMatrix() && ec.getMatrixObject(input2).isFederated(FTypes.FType.ROW) &&
+				ec.getMatrixObject(input2).isFederatedExcept(FTypes.FType.BROADCAST)))
```

Comment:
I am not sure about this, no testcase covers the case explained at the bottom of the comment I think.